src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2446,71 +2452,91 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
             fixed (char* outChars = &MemoryMarshal.GetReference(chars))
             fixed (byte* inData = &MemoryMarshal.GetReference(bytes))
             {
-                charsWritten = ConvertToBase64Array(outChars, inData, 0, bytes.Length, insertLineBreaks);
+                charsWritten = ConvertToBase64Array(outChars, inData, bytes.Length, insertLineBreaks);
                 return true;
             }
         }
 
-        private static unsafe int ConvertToBase64Array(char* outChars, byte* inData, int offset, int length, bool insertLineBreaks)
+        private static int[] CreateBase64PairsTable()
+        {
+            var table = new int[64 * 64];
```

Comment:
```suggestion
            Span<int> table = stackalloc int[64 * 64];
```

I think its okay to stackalloc 4096 bytes

Reply:
No, this table has 'forever' lifetime.
Stack-allocated objects are temporary.
(and `int[64 * 64]` is 16KB)

Reply:
Oh right 🤦🏼 Didn't saw that.

Reply:
How about using `ArrayPool`?

Reply:
But why?
ArrayPool is for Rent/Return usage pattern.

Reply:
Correct. However, it has often turned out that this is faster than allocating 64 * 64 bytes in certain cases.
But not always (i.e. this has to be benchmarked).

You can return the array as soon as you don't need it in `private static unsafe int ConvertToBase64Array(char* outChars, byte* inData, int length, bool insertLineBreaks)`.

So that would be my approach now. But to what extent the ArrayPool is still useful with an array of this size I don't know and you have to benchmark it.

/cc @adamsitnik https://adamsitnik.com/Array-Pool/

Reply:
This method `CreateBase64PairsTable()` invoked only once per process lifetime. So it is not important how optimized it is.